lib/ecto/adapters/myxql/connection.ex

```diff
@@ -572,6 +574,17 @@ if Code.ensure_loaded?(MyXQL) do
       quote_name(literal)
     end
 
+    defp expr({:selected_as, _, [name]}, _sources, _query) do
+      [Atom.to_string(name)]
+    end
+
+    defp expr({:selected_as, _, [_field_expr, _name]}, _sources, _query) do
+      raise ArgumentError,
+          "`selected_as/2` can only be used in the outer most `select` expression. " <>
+            "If you are attempting to alias a field from a subquery, it is not allowed " <>
+              "because subquery fields are automatically aliased by the corresponding map/struct key."
+    end
```

Comment:
Can't we handle this in Ecto? Or this is EctoSQL specific behaviour?

Reply:
I'll have to look into this, but that's a good point. I'll explore this idea before merging.

Reply:
From what I can tell there isn't a really great way to check it in Ecto. When building a query it won't know if `selected_as/2` will eventually go into a subquery or cte. And when collecting the fields in the planner it won't know if it's in a cte/subquery.

The only thing I can think of is to pass a flag into `collect_fields` to say we're in a subquery or cte. @v0idpwn / @josevalim  I can make this change if you agree it's worth it?

Reply:
I am not sure because maybe some databases may allow nested aliases? So I think the enforcing at the root is a SQL specific thing?

Reply:
I think under this situation it might be better here